### PR TITLE
Feature/draft selections

### PR DESCRIPTION
**Pull request by hannyle**
### Description

- Drafts will be shown to users to save as templates before  publishing folders

### Related issues

Closes https://github.com/CSCfi/metadata-submitter-frontend/issues/155

### Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

### Changes Made

- Add a form for selecting drafts in component `WizardDraftSelections`
- Add option to `View` the draft
- Add `Cancel` button to close the `Publish dialog`
- Add e2e test for showing draft selections
- Fix e2e test for error 401

### Testing

- [x] Integration Tests

### Mentions

List of drafts saved as templates will be implemented in the next PR (for issuse https://github.com/CSCfi/metadata-submitter-frontend/issues/141)

### CODE REVIEW

File: src/services/errorMonitor.js [hunk 1]
```diff
@@ -3,7 +3,7 @@ export const errorMonitor = res => {
   if (!res.ok) {
     switch (res.status) {
       case 401:
-        window.location = "/error401"
+        // window.location = "/error401"
```
Comment (blankdots) [inline src/services/errorMonitor.js:6]: forgotten comment ?

File: src/components/NewDraftWizard/WizardComponents/WizardDraftSelections.js [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+import { setSubmissionType } from "features/wizardSubmissionTypeSlice"
+import draftAPIService from "services/draftAPI"
+import type { ObjectInsideFolderWithTags } from "types"
+import { getItemPrimaryText } from "utils"
+
+const useStyles = makeStyles(theme => ({
+  formComponent: {
+    margin: theme.spacing(2),
+    padding: 0,
+    overflow: "scroll",
```
Comment (blankdots) [inline src/components/NewDraftWizard/WizardComponents/WizardDraftSelections.js:29]: image
There is some weird formatting I think we can change this to `auto` and it will be solved, but I have not tested with more items
```suggestion
    overflow: "auto",
```

**Review by blankdots — APPROVED** [review]
small issue other than that looks good, draft is saved under user :+1:

**Comment by hannyle** [comment]
> small issue other than that looks good, draft is saved under user 👍

Looks like it's the responsive css issue since I don't have it on my machine.
You could check if it's fixed with my new commit.

**Review by blankdots — APPROVED** [review]
now it is better

**Comment by lilachic** [comment]
It runs nicely through all the tests. But when selecting study, fill form, give title and select type and save that as draft, then selecting sample and xml and submit, click next and click publish. It'll open sample form to edit instead of study form? And clicking update for empty form gives patch error.

**Comment by blankdots** [comment]
> update for empty form gives patch error.

Sounds like a bug for updating empty form, which i don't think it is related to this issue, but can be filled in as a bug.

**Comment by blankdots** [comment]
although we do have tests for submitting empty form i assume we don't check updating empty form, maybe that bug can be solved as part of this issue :thinking: . any opinions ?

**Comment by lilachic** [comment]
So now is that bug is filed as 245, or is that another?
What is there still left is that when I select a study from, enter title and select type, then select some other object, like sample, select XML file and submit that file, click next and click publish, then opens the select window, then check the study from to edit, click view, it opens sample form instead of study form?

**Comment by blankdots** [comment]
@lilachic that #245 has nothing to do with this issue (i was working on `develop` when i discovered it), if you want to file a bug you can do it or you can talk with @hannyle who opened this PR to understand how it should be solved what you discovered.

**Comment by hannyle** [comment]
> So now is that bug is filed as 245, or is that another?
> What is there still left is that when I select a study from, enter title and select type, then select some other object, like sample, select XML file and submit that file, click next and click publish, then opens the select window, then check the study from to edit, click view, it opens sample form instead of study form?

@lilachic I understand what you have mentioned. There are 2 issues here: 
- `View` draft redirects to a wrong `objectType`
- Empty form issue https://github.com/CSCfi/metadata-submitter-frontend/issues/245

I have pushed some fixes, please check again if if works correctly now.

File: src/components/NewDraftWizard/WizardComponents/WizardAlert.js [hunk 3]
```diff
@@ -92,6 +95,7 @@ const CancelFormDialog = ({
   }
 
   let [dialogTitle, dialogContent] = ["", ""]
+  // let draftSelections = null
```
Comment (lilachic) [inline src/components/NewDraftWizard/WizardComponents/WizardAlert.js:98]: Is the comment needed?

**Review by lilachic — APPROVED** [review]
Now it looks good.  Seems to work correctly.

**Comment by hannyle** [comment]
> although we do have tests for submitting empty form i assume we don't check updating empty form, maybe that bug can be solved as part of this issue 🤔 . any opinions ?

I can add the test for `Update` empty form as part of issue `245`
